features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/sha/sha1_alt.c

```diff
@@ -49,15 +54,21 @@ void mbedtls_sha1_free(mbedtls_sha1_context *ctx)
     if (ctx->ishw) {
         mbedtls_sha1_hw_free(&ctx->hw_ctx);
         crypto_sha_release();
-    }
-    else {
+    } else {
         mbedtls_sha1_sw_free(&ctx->sw_ctx);
     }
```

Comment:
Currently, it's not safe to call `mbedtls_sha1_free` multiple times, which generally it is. One way to solve this would be to rename `is_hw` to `initialized_as` (or similar), assuming three valid values `0` for uninitialized, `1` for SW initialized, and `2` for HW initialized. Then `sha1_free` could reset it to `0` after appropriate cleanup, and return immediately if `initialized_as` is already `0`.

Reply:
@hanno-arm By following your idea, I reworked it in https://github.com/ARMmbed/mbed-os/pull/4925/commits/0b58d60dfb279048ee62cb09aa42e39f0b5ff08a.